test/integration/suites/k8s/conf/agent/spire-agent.yaml

```diff
@@ -111,6 +111,10 @@ spec:
       hostNetwork: true
       dnsPolicy: ClusterFirstWithHostNet
       serviceAccountName: spire-agent
+      # Make sure that we can create the directory for the socket in the host
```

Comment:
Why does only this one test need to run as root to create the socket directory? Could this be avoided with a config change in the test?

Reply:
This is needed because we use a hostPath volume to share the socket for the Workload API. I've moved the `securityContext` field to affect the spire-agent container only.